modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd

```diff
@@ -140,6 +140,22 @@
     < 'STRING', 'URI', 'BINARY', 'LONG', 'DOUBLE', 'DECIMAL', 'BOOLEAN', 
       'DATE', 'NAME', 'PATH', 'REFERENCE', 'WEAKREFERENCE', 'UNDEFINED'
 
+// -----------------------
+// Large unordered collections
+// -----------------------
+// Abstract base type for all unordered collections.
+
+[mode:unorderedCollection] abstract
+  + * (nt:base) = nt:unstructured abort 
```

Comment:
I guess we can use mixins, as long as we enforce constraints like: 
a) no unordered collection mixin can be added or removed if the node on which it is added/removed has any children.
b) if a node already has such a mixin, a second one is not allowed
That means we could a) remove the base `mode:unorderedCollection` type and b) transform the 4 subtypes into mixins.
@rhauch thoughts ?